File: backend/services/image_service.py
from database import get_connection


def add_image_match(image_name, user_id):
    """
    Add a new image match to the database.

    Args:
        image_name (str): The name of the image
        user_id (int): The ID of the user associated with this image

    Returns:
        dict: A dictionary containing:
            - image_id: The newly created image ID
            - image_name: The image name
            - user_id: The user ID
    """
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        'INSERT INTO image_match (image_name, user_id) VALUES (?, ?)',
        (image_name, user_id)
    )
    conn.commit()
    image_id = cursor.lastrowid
    conn.close()

    return {
        'image_id': image_id,
        'image_name': image_name,
        'user_id': user_id
    }


def get_images_by_user(user_id):
    """
    Get all images associated with a specific user.

    Args:
        user_id (int): The ID of the user

    Returns:
        list: A list of dictionaries, each containing:
            - image_id: The image ID
            - image_name: The image name
            - user_id: The user ID
    """
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute(
        'SELECT image_id, image_name, user_id FROM image_match WHERE user_id = ?',
        (user_id,)
    )
    rows = cursor.fetchall()
    conn.close()

    images = []
    for row in rows:
        images.append({
            'image_id': row[0],
            'image_name': row[1],
            'user_id': row[2]
        })

    return images

File: frontend/src/services/ImageService.ts
const API_BASE_URL = import.meta.env.VITE_BACKEND_BASE_API_URL || 'http://localhost:8000';

export interface AddImageMatchRequest {
  image_name: string;
  user_id: number;
}

export interface AddImageMatchResponse {
  image_id: number;
  image_name: string;
  user_id: number;
  status: string;
}

export interface GetUserImagesRequest {
  user_id: number;
}

export interface ImageMatch {
  image_id: number;
  image_name: string;
  user_id: number;
}

export interface GetUserImagesResponse {
  user_id: number;
  images: ImageMatch[];
  count: number;
  status: string;
}

/**
 * Add a new image match to the database.
 * Associates an image name with a user ID.
 */
export const addImageMatch = async (
  image_name: string,
  user_id: number
): Promise<AddImageMatchResponse> => {
  try {
    const response = await fetch(`${API_BASE_URL}/api/add-image-match`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ image_name, user_id }),
    });

    if (!response.ok) {
      const errorData = await response.json().catch(() => ({}));
      throw new Error(errorData.detail || `HTTP error! status: ${response.status}`);
    }

    const data: AddImageMatchResponse = await response.json();
    return data;
  } catch (error) {
    console.error('Error calling add-image-match:', error);
    throw error;
  }
};

/**
 * Get all images associated with a specific user.
 * Returns a list of all image_ids and image_names for the given user_id.
 */
export const getUserImages = async (user_id: number): Promise<GetUserImagesResponse> => {
  try {
    const response = await fetch(`${API_BASE_URL}/api/get-user-images`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ user_id }),
    });

    if (!response.ok) {
      const errorData = await response.json().catch(() => ({}));
      throw new Error(errorData.detail || `HTTP error! status: ${response.status}`);
    }

    const data: GetUserImagesResponse = await response.json();
    return data;
  } catch (error) {
    console.error('Error calling get-user-images:', error);
    throw error;
  }
};
